core/src/main/java/gnu/trove/THashMap.java

```diff
@@ -743,6 +743,32 @@ public V setValue(V o) {
                                 // setValue is called again
             return prev;
         }
+
+
+        @Override
+        public final boolean equals(Object o) {
+            if (o == this) {
+                return true;
+            }
+            if (o instanceof Map.Entry) {
+                Map.Entry<?, ?> otherEntry = (Map.Entry<?, ?>) o;
+                Object otherKey = otherEntry.getKey();
+                Object otherValue = otherEntry.getValue();
+                //noinspection unchecked
+                if (!(key == otherKey || key != null && _hashingStrategy.equals(key, (K) otherKey))) {
+                    return false;
+                }
+                return val == otherValue || (val != null && val.equals(otherValue));
+            }
+            return false;
+        }
+
+        @Override
+        public final int hashCode() {
+            HashProcedure hashProcedure = new HashProcedure();
+            hashProcedure.execute(key, val);
```

Comment:
The `Map.Entry.hashCode` is also fully specified:

> Returns the hash code value for this map entry. The hash code of a map entry e is defined to be:
> ```java
>               (e.getKey()==null   ? 0 : e.getKey().hashCode()) ^
>               (e.getValue()==null ? 0 : e.getValue().hashCode())
> ```           
> This ensures that `e1.equals(e2)` implies that `e1.hashCode()==e2.hashCode()` for any two Entries `e1` and `e2`, as required by the general contract of `Object.hashCode`.

I don't think the `HashProcedure` will produce the same algorithm. Please write it according to the spec:

```java
int keyHash = (key==null ? 0 : key.hashCode());
int valueHash = (val==null ? 0 : val.hashCode());
return keyHash ^ valueHash;
```